File: php/f30.loop.php
#!/usr/bin/php
<?php
// spin loop as always true
//   break used to exit loop
do {
    echo "Enter your name (quit to Exit): "; # output prompt
    $answer = rtrim(fgets(STDIN));           # get input, trim newline
    if ($answer == "quit")
         break;                              # exit loop if user quits
    else
        echo "Hello $answer!\n";             # output results as not exiting
} while (1);
?>

File: python/f30.loop.py
#!/usr/bin/python
# spin loop as always true condition
#   break to exit from loop
while True:
  # print a prompt and get answer
  answer = raw_input("Enter your name (quit to Exit): ")
  if answer == "quit":
    break                      # exit loop as user wants to quit
  else:
    print "Hello %s!" % answer # output result as we are not exiting